File: Challenge 12.py
'''
Write a program that will display all the factors of a number, entered by the user, that are bigger than 1.
(e.g. the factors of the number 12 are 6,4,3 and 2 because they divide into 12 exactly).
Hint
to find out whether a number X is a factor of Y use :
IF Y mod X =0 (there is nothing remaining when Y is divided by X)

Extension
Tell the user if the number they
entered is a prime number 
'''

num = int(input('Factors of what number do you want to know? '))
flag = False

if num > 1:
    for i in range(2, num):
        if (num % i) == 0:
            flag = True
            break

if flag:
    print(num, "is not a prime number")
else:
    print(num, "is a prime number")

def number_factors(num):
    factors = []
    for i in range(2, num):
        if num % i == 0:
            factors.append(i)
    print('Factors: ', factors)

number_factors(num)

File: Challenge 12.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

class NumberFactors {
    static int getUserIntInput() {
        boolean check = false;
        int output = 34404;
        System.out.print(">> ");
        Scanner sc_input = new Scanner(System.in);
        while (check == false) {
            String usr_input = sc_input.nextLine();
            try {
                int number = Integer.parseInt(usr_input);
                if (number > 2) {
                    output = number;
                    check = true;
                } else {
                    System.out.println("You need to input number bigger than 2.");
                }
            } catch (NumberFormatException e) {
                System.out.println("You need to input a number.");
            }
        }
        return output;
    }

    static void printNumberFactors(int num) {
        List<Integer> factors = new ArrayList<>();
        for (int i = 2; i <= num; i++) {
            if (num % i == 0) {
                factors.add(i);
            }
        }
        for (int i: factors) {
            System.out.print(i + ", ");
        }
    }

    public static void main(String[] args) {
        System.out.println("Factors of what number do you want to know?");
        int usr_number = getUserIntInput();
        System.out.println("Factor of number " + usr_number + ":");
        printNumberFactors(usr_number);
    }
}
